src/pages/CustomerInvoicePage.js

```diff
@@ -35,56 +34,32 @@ const MODAL_KEYS = {
 export class CustomerInvoicePage extends GenericPage {
   constructor(props) {
     super(props);
-    this.state.sortBy = 'itemName';
-    this.state.columns = [
-      {
-        key: 'itemCode',
-        width: 2,
-        title: tableStrings.item_code,
-        sortable: true,
-      },
-      {
-        key: 'itemName',
-        width: 4,
-        title: tableStrings.item_name,
-        sortable: true,
-      },
-      {
-        key: 'availableQuantity',
-        width: 2,
-        title: tableStrings.available_stock,
-        sortable: true,
-        alignText: 'right',
-      },
-      {
-        key: 'totalQuantity',
-        width: 2,
-        title: tableStrings.quantity,
-        sortable: true,
-        alignText: 'right',
-      },
-      {
-        key: 'remove',
-        width: 1,
-        title: tableStrings.remove,
-        alignText: 'center',
-      },
-    ];
-    this.dataTypesSynchronised = DATA_TYPES_SYNCHRONISED;
-    this.finalisableDataType = 'Transaction';
-    this.getFilteredSortedData = this.getFilteredSortedData.bind(this);
-    this.onAddMasterItems = this.onAddMasterItems.bind(this);
-    this.openItemSelector = this.openItemSelector.bind(this);
-    this.openCommentEditor = this.openCommentEditor.bind(this);
-    this.openTheirRefEditor = this.openTheirRefEditor.bind(this);
-    this.getModalTitle = this.getModalTitle.bind(this);
-    this.renderPageInfo = this.renderPageInfo.bind(this);
+    this.state = {
+      selection: [],
+      modalKey: null,
+      modalIsOpen: false,
+    };
+    this.dataFilters = {
+      searchTerm: '',
+      sortBy: 'itemName',
+      isAscending: true,
+    };
+    autobind(this);
+  }
+
+  updateDataFilters(newSearchTerm, newSortBy, newIsAscending) {
+    // We use != null, which checks for both null or undefined (undefined coerces to null)
+    if (newSearchTerm != null) this.dataFilters.searchTerm = newSearchTerm;
```

Comment:
Also sortDataBy function.
I see the case for manual sorting, it's when the field is not in a realm schema. So let's say we are trying to use realm to sort otherPartyName (which is getter) we will get -> property doesn't exist (I assume realm does `Object.getOwnPropertyDescriptor` check before sorting, which will return undefined for getters, unlike `typeof` which will return `string` for otherPartyName).. I would prefer us using realm as much as possible, so in sortDataBy, it would be good to:

```
// check !data or data.length < 1, return []
// check data.length < 1 return []
// check !Object.getOwnPropertyDescriptor(data[0], sortBy), return manualSort()
```
it would also be nice if manualSort can figure out sortDatType from schema, but I see some problems with currentScheme, i.e serialNumber is a string.

Not to be hypocritical and to enforce consistency (using realm as much as possible), Transaction.removeTransactionBatchesById should be refactored to use filtered().

I think this can be done in another PR, once this one is merged I will create